homeassistant/components/noaa_tides/sensor.py

```diff
@@ -51,24 +51,31 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     else:
         unit_system = UNIT_SYSTEMS[0]
 
-    noaa_sensor = NOAATidesAndCurrentsSensor(name, station_id, timezone, unit_system)
+    station = coops.Station(station_id, unit_system)
```

Comment:
Is IO done here?  Seems probable based on the check below.  Should you also catch exceptions on this line?

Reply:
Yes, you're right, it fetches metadata for the station. I will catch the `KeyError` that gets thrown if the station_id does not exist - I'll add a commit.

Should I also catch general IO/connection issues or does home assistant take care of that for me? Quickly browsing the repository, it looks like some components do and others don't. Would you suggest something like `requests.exceptions.ConnectionError` (the noaa-coops python package uses requests to fetch data)?

Thank you for the comments; I appreciate your help. I've created a couple custom components for my own setup but this is my first official home assistant PR.

Reply:
If you don't catch connection errors, then your platforms won't set up if there is one.

> requests.exceptions.ConnectionError

Makes sense to me.  You could also consider reworking this interegration to use DataUpdateCoordinator.

Reply:
Gotcha, I'll just catch with ConnectionError for now. There are a couple improvements I'd like to make eventually but, since this integration hasn't been working for almost a month now, I want to keep this PR as small/simple as possible.